xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/BaseElement.java

```diff
@@ -101,7 +109,28 @@ public abstract class BaseElement<R extends EntityReference> implements ElementI
     private EntityReferenceSerializer<String> localUidStringEntityReferenceSerializer;
 
     private ContextualLocalizationManager localization;
-    
+
+    /**
+     * @return true of the element was modified (or created)
+     * @since 17.1.0RC1
+     */
+    public boolean isDirty()
+    {
+        return this.dirty;
+    }
+
+    /**
+     * @param dirty true of the element was modified (or created)
+     * @since 17.1.0RC1
+     */
+    public void setDirty(boolean dirty)
+    {
+        this.dirty = dirty;
+        if (dirty && this.ownerDocument != null) {
+            this.ownerDocument.setMetaDataDirty(true);
```

Comment:
Shouldn't this method be overridden in `BaseProperty` to also mark the object as dirty?

Reply:
I do remember thinking I needed to do that, but looks like it then slipped my mind.